### PR TITLE
Rest API Implementation

### PR DESCRIPTION
**Pull request by MasonHN**

### CODE REVIEW

**Comment by kev-chan** [comment]
also please outline the file/folder structure to match what is expected with dropwizard

File: src/main/java/com/twitter/resources/TwitterClientResource.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            return Response.status(Response.Status.OK).entity(new Timeline(statuses)).build();
+        }
+        catch(TwitterException e){
+            e.printStackTrace();
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity("ERROR: Please try your request again later").build();
+        }
+    }
+    @POST
+    @Path("/tweet/{id}")
+    public Response createTweet(@PathParam("id") String tweet) {
```
Comment (kev-chan) [inline src/main/java/com/twitter/resources/TwitterClientResource.java]: this seems pretty odd if you had a long tweet

Comment (kev-chan) [reply src/main/java/com/twitter/resources/TwitterClientResource.java]: change this from pathParam as having the tweet on the path doesn't make sense with spaces.  Take a look at Post in particular and what is available there

File: src/main/java/com/twitter/resources/TwitterClientResource.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        }
+    }
+
+    @POST
+    @Path("/tweet")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response createTweet(Tweet body) {
+        String tweet = body.tweet;
+        if(tweet.length() > 280) {
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity("ERROR: Please make sure your tweet is 280 characters or less.").build();
```
Comment (kev-chan) [inline src/main/java/com/twitter/resources/TwitterClientResource.java]: you don't need to state ERROR as that's implied through the status code.

File: src/main/java/com/twitter/resources/TwitterClientResource.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity("ERROR: Please try your request again later").build();
+        }
+    }
+
+    @POST
+    @Path("/tweet")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response createTweet(Tweet body) {
+        String tweet = body.tweet;
+        if(tweet.length() > 280) {
```
Comment (kev-chan) [inline src/main/java/com/twitter/resources/TwitterClientResource.java]: pull out the value so it isn't hardcoded within the method

File: src/main/java/com/twitter/resources/TwitterClientResource.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        }
+        try {
+            Twitter twitter = TwitterFactory.getSingleton();
+            Status status = twitter.updateStatus(tweet);
+            System.out.println("Successfully tweeted \"" + status.getText() + "\"");
+            return Response.status(Response.Status.OK).entity(status).build();
+        }
+        catch(TwitterException e){
+            e.printStackTrace();
+            if (e.getStatusCode() == 403) {
```
Comment (kev-chan) [inline src/main/java/com/twitter/resources/TwitterClientResource.java]: use existing enum vs 403

File: src/main/java/com/twitter/api/Tweet.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package com.twitter.api;
```
Comment (kev-chan) [inline src/main/java/com/twitter/api/Tweet.java]: why is this called api>

Comment (MasonHN) [reply src/main/java/com/twitter/api/Tweet.java]: In the drop wizard docs they label the directory with all their representation classes as api. So I was just going off that.

File: src/main/java/com/twitter/api/Tweet.java [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class Tweet {
+
+    @JsonProperty("tweet")
+    public String tweet;
+
+    public static final int maxSize = 280;
```
Comment (kev-chan) [inline src/main/java/com/twitter/api/Tweet.java]: This probably shouldn't be here as it's odd to have this inside the java object.  I'd recommend either keeping it in the resource class or in a separate utils class

File: src/main/java/com/twitter/resources/TwitterClientResource.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        }
+    }
+
+    @POST
+    @Path("/tweet")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response createTweet(Tweet body) {
+        String tweet = body.tweet;
+        if(tweet.length() > Tweet.maxSize) {
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity("Please make sure your tweet is 280 characters or less.").build();
```
Comment (kev-chan) [inline src/main/java/com/twitter/resources/TwitterClientResource.java]: probably better to use the constant for the 280 here so it's consistent.

**Pull request merged**